lte/gateway/c/connection_tracker/CMakeLists.txt

```diff
@@ -0,0 +1,132 @@
+# limitations under the License.
+
+cmake_minimum_required(VERSION 3.0.2)
+
+PROJECT(ConnectionTracker)
+
+add_compile_options(-std=c++14 -g -O0)
+
+include($ENV{MAGMA_ROOT}/orc8r/gateway/c/common/CMakeProtoMacros.txt)
+
+if (NOT BUILD_TESTS)
+  # Add AddressSanitizer (asan) support for debug builds of SessionD
+  set (CMAKE_CXX_FLAGS_DEBUG
+      "${CMAKE_CXX_FLAGS_DEBUG} -fno-omit-frame-pointer -fsanitize=address")
+  set (CMAKE_LINKER_FLAGS_DEBUG
+      "${CMAKE_LINKER_FLAGS_DEBUG} -fno-omit-frame-pointer -fsanitize=address")
+  # Add LeakSanitizer (lsan) support to the release build of SessionD so that
+  # we can find memory leaks in production.
+  set(CMAKE_C_FLAGS_RELWITHDEBINFO
+     "${CMAKE_C_FLAGS_RELWITHDEBINFO} -fsanitize=leak -fno-omit-frame-pointer")
+  set(CMAKE_EXE_LINKER_FLAGS_RELWITHDEBINFO
+     "${CMAKE_EXE_LINKER_FLAGS_RELWITHDEBINFO} -fsanitize=leak")
+endif ()
+
+set(OUTPUT_DIR "${CMAKE_CURRENT_BINARY_DIR}")
+
+set(MAGMA_LIB_DIR $ENV{C_BUILD}/magma_common)
+add_definitions(-DLOG_WITH_GLOG)
+
+message("Build type is ${CMAKE_BUILD_TYPE}")
+if (CMAKE_BUILD_TYPE STREQUAL "Debug")
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Werror -Wno-unused-function")
+endif ()
+
+include_directories("${OUTPUT_DIR}")
+include_directories("${MAGMA_ROOT}/orc8r/gateway/c/common/logging")
+
+
+include_directories(${MAGMA_LIB_DIR}/service303)
+include_directories(${MAGMA_LIB_DIR}/service_registry)
+include_directories(${MAGMA_LIB_DIR}/async_grpc)
+include_directories(${MAGMA_LIB_DIR}/config)
+include_directories(${MAGMA_LIB_DIR}/datastore)
+include_directories(${MAGMA_LIB_DIR}/policydb)
+include_directories(${MAGMA_LIB_DIR}/eventd)
+
+include($ENV{MAGMA_ROOT}/orc8r/gateway/c/common/CMakeProtoMacros.txt)
+
+# compile the needed macros
+create_proto_dir("lte" LTE_CPP_OUT_DIR)
+create_proto_dir("orc8r" ORC8R_CPP_OUT_DIR)
+create_proto_dir("feg/gateway/services/aaa" CWF_CPP_OUT_DIR)
+
+list(APPEND PROTO_SRCS "")
+list(APPEND PROTO_HDRS "")
+
+message("Proto_srcs are ${PROTO_SRCS}")
+
+find_library(SERVICE303_LIB SERVICE303_LIB HINTS ${MAGMA_LIB_DIR}/service303)
+
+# compile the needed macros
+create_proto_dir("lte" LTE_CPP_OUT_DIR)
+create_proto_dir("orc8r" ORC8R_CPP_OUT_DIR)
+create_proto_dir("feg/gateway/services/aaa" CWF_CPP_OUT_DIR)
+
+list(APPEND PROTO_SRCS "")
+list(APPEND PROTO_HDRS "")
+
+
+set(SMGR_LTE_CPP_PROTOS session_manager subscriberdb policydb
+  pipelined spgw_service mconfig/mconfigs)
+generate_cpp_protos("${SMGR_LTE_CPP_PROTOS}" "${PROTO_SRCS}"
+  "${PROTO_HDRS}" ${LTE_PROTO_DIR} ${LTE_CPP_OUT_DIR})
+
+set(SMGR_CWF_CPP_PROTOS accounting context)
+generate_cwf_cpp_protos("${SMGR_CWF_CPP_PROTOS}" "${PROTO_SRCS}"
+  "${PROTO_HDRS}" ${CWF_PROTO_DIR} ${CWF_CPP_OUT_DIR})
+
+message("Proto_srcs are ${PROTO_SRCS}")
+
+
+link_directories(
+  ${MAGMA_LIB_DIR}/async_grpc
+  ${MAGMA_LIB_DIR}/config
+  ${MAGMA_LIB_DIR}/service303
+  ${MAGMA_LIB_DIR}/datastore
+  ${MAGMA_LIB_DIR}/policydb
+  ${MAGMA_LIB_DIR}/eventd
```

Comment:
is there plans to add event logging to these? not sure why this was added since I couldn't find usages for this